text/src/autogluon/text/automm/data/preprocess_dataframe.py

```diff
@@ -175,8 +175,9 @@ def fit(
             elif col_type == CATEGORICAL:
                 if self._config["categorical"]["convert_to_text"]:
                     # Convert categorical column as text column
-                    processed_data = col_value.apply(lambda ele: '' if ele is None else str(ele))
-                    if len(np.unique(processed_data)) == 1:
+                    col_value = col_value.astype("object")
+                    processed_data = col_value.apply(lambda ele: '' if pd.isnull(ele) else str(ele))
```

Comment:
Note: This is automatically handled in feature generator (text NaN -> empty str)

Reply:
Ditto. Will change after API alignment.